sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -587,10 +592,10 @@ def run_ParDo(self, transform_node):
     si_labels = {}
     full_label_counts = defaultdict(int)
     lookup_label = lambda side_pval: si_labels[side_pval]
-    for ix, side_pval in enumerate(transform_node.side_inputs):
+    for side_pval in transform_node.side_inputs:
       assert isinstance(side_pval, AsSideInput)
       step_name = 'SideInput-' + self._get_unique_step_name()
-      si_label = 'side%d' % ix
+      si_label = self._get_unique_side_input_name()
```

Comment:
As I mentioned, it'd be cleaner to let this be step name + ix rather than introducing global state.